SharpBucket/V2/EndPoints/ListIssuesParameters.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using SharpBucket.V2.Pocos;
+
+namespace SharpBucket.V2.EndPoints
+{
+    /// <summary>
+    /// An object that can be passed on methods that list issues.
+    /// </summary>
+    public class ListIssuesParameters : ListParameters
+    {
+        /// <summary>
+        /// The list of the kinds of the issues to enumerate.
+        /// If null or empty, it defaults to listing all issues.
+        /// </summary>
+        public IReadOnlyCollection<IssueKind> Kinds { get; set; }
+
+        /// <summary>
+        /// The list of the priorities of the issues to enumerate.
+        /// If null or empty, it defaults to listing all issues.
+        /// </summary>
+        public IReadOnlyCollection<IssuePriority> Priorities { get; set; }
+
+        /// <summary>
+        /// The list of the states of the issues to enumerate.
+        /// If null or empty, it defaults to listing all issues.
+        /// </summary>
+        public IReadOnlyCollection<IssueStatus> Statuses { get; set; }
```

Comment:
Where/How did you find that parameters ?
Because the API documentation says nothing about that: https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Bworkspace%7D/%7Brepo_slug%7D/issues#get

The issues endpoint is effectively listed in the endpoints that support filtering: https://developer.atlassian.com/bitbucket/api/2/reference/meta/filtering
Which is correctly represented here by extending ListParameters, but, I'm not sure that there is effectively another way than the 'q' parameter (for Query) and its particular syntax to filter this endpoint like you seem to expose here.

So maybe you should just use ListParameters and don't create a new type dedicated to Issues ?

Reply:
I'm not certain where exactly I found it, but I did do a lot of digging around with the issues themselves and this link along with their Advanced search supports this:
https://bitbucket.org/penev92/bottestrepository/issues?status=new&status=open
According to the Advanced search you can filter by a bunch of different parameters:
![image](https://user-images.githubusercontent.com/7137365/76691120-f362a800-664f-11ea-9200-d0044551708a.png)

Upon further testing it doesn't seem to be working. That sounds like a great oversight on BitBucket's part or am I missing something?

Reply:
All that filters can be done through the filtering feature, which is a string that should be set in the Filter property of the ListParameters or EnumerateParameters in SharpBucket.
We have start a FilterBuilder class, but it's really small, and not really advanced.

On BitBucket side the filter should be defined in the query string as 'q' parameter.
For the syntax you can look at the page I mention:
https://developer.atlassian.com/bitbucket/api/2/reference/meta/filtering

Furthermore, what you did, was just able to say: "Status" "New", while has you can see on the screenshot you provide we should says: "Status" "is" "new". The filtering feature need to known the operator to use to compare the property with the values...

Reply:
I guess I'll need help with this since using `ListParameters` with `Filter = "status=open"` results in a 
```
SharpBucket.V2.BitbucketV2Exception : Invalid filter query expression:
Syntax error in input at 'open' (type FIELD)
```
because it tries to do `q=status=open`. The syntax here should be identical to PullRequest filtering, that's why I tried to mirror `ListPullRequestsParameters`.

Reply:
You should try
`Filter = "status=\"open\""`
With double quotes around the string value for status

or using the FilterBuilder class that I mention:
`Filter = FilterBuilder.ParseSingleQuotedString("status='open'")`
If it seems more readable for you with single quoted strings

by the way sorry for the delay, I was occupied with family matters recently, but I should be more reactive now.